user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -26,73 +26,119 @@
+static void update_setpoints(Gimbal_t *gimbal);
+static void increment_PID(Gimbal_t *gimbal);
+
 
 /**
- * @brief Initialises PID and fetches Gimbal motor data to ensure 
+ * @brief Initializes PID and fetches Gimbal motor data to ensure 
  *  gimbal points in direction specified. 
  * @param FreeRTOS parameters
  * @retval None
  */
 void gimbal_task(void* parameters){
 
-    vTaskDelay(GIMBAL_TASK_INIT_TIME);
-	
-    gimbal.pitch_motor->gimbal_motor_raw = get_Pitch_Gimbal_Motor_Measure_Point();
-    gimbal.yaw_motor->gimbal_motor_raw = get_Yaw_Gimbal_Motor_Measure_Point();
-
-    fp32 pid_constants[3] = {pid_kp, pid_ki, pid_kd};
-    PidTypeDef pid_pitch;
-    PID_Init(&pid_pitch, PID_POSITION, pid_constants, max_out, max_iout);
-
-    PidTypeDef pid_yaw;
-    PID_Init(&pid_yaw, PID_POSITION, pid_constants, max_out, max_iout);
-    
-
-    rc_ptr = get_remote_control_point();
+    vTaskDelay(200);
+	initialization(&gimbal);
     
     while(1){	
-        /* For now using strictly encoder feedback for position */
-     
+        //send_to_uart(&gimbal);
         
-        // Get CAN received data
-        gimbal.pitch_pos_raw = gimbal.pitch_motor->gimbal_motor_raw->ecd;
-        gimbal.pitch_speed_raw = gimbal.pitch_motor->gimbal_motor_raw->speed_rpm;
-
-        gimbal.yaw_pos_raw = gimbal.yaw_motor->gimbal_motor_raw->ecd;
-        gimbal.yaw_speed_raw = gimbal.yaw_motor->gimbal_motor_raw->speed_rpm;
-
-        // Calculate setpoints based on RC signal.
-        if(switch_is_mid(rc_ptr->rc.s[0]) || switch_is_up(rc_ptr->rc.s[0])){
-            gimbal.yaw_pos_set = linear_map_int_to_int(rc_ptr->rc.ch[2], RC_MIN, RC_MAX, YAW_MAX, YAW_MIN);
-            gimbal.pitch_pos_set = linear_map_int_to_int(rc_ptr->rc.ch[3], RC_MIN, RC_MAX, PITCH_MAX, PITCH_MIN);
-        }else{
-            gimbal.yaw_pos_set = linear_map_int_to_int(0, RC_MIN, RC_MAX, YAW_MAX, YAW_MIN);
-            gimbal.pitch_pos_set = linear_map_int_to_int(0, RC_MIN, RC_MAX, PITCH_MAX, PITCH_MIN);
-        }
+        /* For now using strictly encoder feedback for position */
         
-        gimbal.pitch_speed_set = PID_Calc(&pid_pitch, gimbal.pitch_motor->pos_raw, gimbal.pitch_pos_set);
-        gimbal.yaw_speed_set = PID_Calc(&pid_yaw, gimbal.yaw_motor->pos_raw, gimbal.yaw_pos_set);
-
+        get_new_data(&gimbal);
+        //send_to_uart(&gimbal);
+        update_setpoints(&gimbal);
+        //send_to_uart(&gimbal);
+        increment_PID(&gimbal);
+        //send_to_uart(&gimbal);
         // Turn gimbal motor
-        CAN_CMD_GIMBAL(gimbal.yaw_speed_set, gimbal.pitch_speed_set, 0, 0);
         
-        vTaskDelay(1);
         
-        //send_to_uart(gimbal_pitch_motor, pid_pitch, pitch_signal);  //Sending data via UART
+        CAN_CMD_GIMBAL( (int16_t) gimbal.yaw_motor.current_out, 
+                        (int16_t) gimbal.pitch_motor.current_out,
+                        0,
+                        0);
+        
+          //Sending data via UART
+        vTaskDelay(5);
```

Comment:
We should use defines for vTaskDelay duration. Also gimbal task delay should be 1, for purpose of shoot integration.

Reply:
Ok